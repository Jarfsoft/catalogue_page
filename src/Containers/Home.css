.poke-section {
  text-align: center;
  background-color: #282c34;
  padding: 2rem 0;
}

.poke-grid {
  color: black;
  display: grid;
  grid-template-columns: auto auto auto;
}

.grid-buttons {
  background-color: white;
  padding: 4rem 0;
  border-radius: 15px;
  margin: 2rem ;
}

.grid-buttons button {
  color: white;
  font-weight: bold;
  font-size: large;
  margin: 0 15rem;
  padding: 0.5rem;
  background-color: rgb(169, 0, 0);
  border: outset 2px rgb(128, 0, 0);
  border-radius: 15px;
}
