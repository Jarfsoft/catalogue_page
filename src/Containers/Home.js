import React from 'react';
import { useSelector, useDispatch } from 'react-redux';

import actions from '../Actions/index';
import { getPokemons } from '../pokeApi';
import Pokemon from '../Components/Pokemon';
import './Home.css';

function Home() {
  const dispatch = useDispatch();

  const list = useSelector((state) => state.pokemons.list);
  const sType = useSelector((state) => state.pokemons.type);
  const continuous = () => {
    let cont = true;
    for (let i = 0; i < 11; i += 1) {
      if (list[i] !== (list[i + 1] - 1)) cont = false;
    }
    return cont;
  };
  const setInfo = (list, type) => {
    const newState = { list, type };
    dispatch(actions.filter(newState));
  };
  const nextClickHandler = () => {
    if (list.length === 12) {
      if (continuous()) {
        dispatch(actions.next(12));
      } else {
        getPokemons(list[11] + 1, sType, setInfo);
      }
    }
  };
  const previousClickHandler = () => {
    if (list.length === 12) {
      if (continuous()) {
        dispatch(actions.next(-12));
      } else {
        getPokemons(1, sType, setInfo);
      }
    }
  };
  return (
    <div className="poke-section">
      <div className="grid-buttons">
        <button onClick={previousClickHandler} type="button">Previous</button>
        <button onClick={nextClickHandler} type="button">Next</button>
        <div className="poke-grid">
          {list ? list.map((p) => <Pokemon key={p} id={p} />) : ''}
        </div>
        <button onClick={previousClickHandler} type="button">Previous</button>
        <button onClick={nextClickHandler} type="button">Next</button>
      </div>
    </div>
  );
}

export default Home;
