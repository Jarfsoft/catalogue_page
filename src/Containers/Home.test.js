import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider, useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import Home from './Home';

const initialState = {
  list: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
  type: '',
};

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn()
    .mockReturnValueOnce(initialState),
}));

describe('Home page render', () => {
  afterEach(() => {
    useSelector.mockClear();
  });
  const mockStore = configureStore();
  let store;

  it('Home page is rendered', () => {
    store = mockStore(initialState);
    const home = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </Provider>,
      ).toJSON();
    expect(home).toMatchSnapshot();
  });

  describe('Home contains', () => {
    it('should contain 4 buttons', () => {
      const home = render(<Provider store={store}><Home /></Provider>);
      const select = (home.container.querySelectorAll('button'));
      expect(select).toHaveLength(4);
    });
  });
});
