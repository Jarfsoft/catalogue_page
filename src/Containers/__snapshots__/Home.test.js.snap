// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page render Home page is rendered 1`] = `
<div
  className="poke-section"
>
  <div
    className="grid-buttons"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Previous
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
    <div
      className="poke-grid"
    >
      
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Previous
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
